doc/source/user_guide/user_defined_functions.rst

```diff
@@ -0,0 +1,336 @@
+| Method                     | Function Input         | Function Output          | Description                                                               |
++============================+========================+==========================+===========================================================================+
+| :meth:`map`                | Scalar                 | Scalar                   | Apply a function to each element                                          |
++----------------------------+------------------------+--------------------------+---------------------------------------------------------------------------+
+| :meth:`apply` (axis=0)     | Column (Series)        | Column (Series)          | Apply a function to each column                                           |
++----------------------------+------------------------+--------------------------+---------------------------------------------------------------------------+
+| :meth:`apply` (axis=1)     | Row (Series)           | Row (Series)             | Apply a function to each row                                              |
++----------------------------+------------------------+--------------------------+---------------------------------------------------------------------------+
+| :meth:`agg`                | Series/DataFrame       | Scalar or Series         | Aggregate and summarizes values, e.g., sum or custom reducer              |
++----------------------------+------------------------+--------------------------+---------------------------------------------------------------------------+
+| :meth:`transform`          | Series/DataFrame       | Same shape as input      | Apply a function while preserving shape; raises error if shape changes    |
++----------------------------+------------------------+--------------------------+---------------------------------------------------------------------------+
+| :meth:`filter`             | -                      | -                        | Return rows that satisfy a boolean condition                              |
++----------------------------+------------------------+--------------------------+---------------------------------------------------------------------------+
+| :meth:`pipe`               | Series/DataFrame       | Series/DataFrame         | Chain functions together to apply to Series or Dataframe                  |
++----------------------------+------------------------+--------------------------+---------------------------------------------------------------------------+
+
+.. note::
+    Some of these methods are can also be applied to groupby, resample, and various window objects.
+    See :ref:`groupby`, :ref:`resample()<timeseries>`, :ref:`rolling()<window>`, :ref:`expanding()<window>`,
+    and :ref:`ewm()<window>` for details.
+
+
+Choosing the Right Method
```

Comment:
Personally I'd only keep the first table. I think the description can already explain anything not clear present in the second table.

And I'd make this section `choosing the right method` part of the `Methods that support User-Defined Functions`. To me it feels like we're having two sections for mostly the same.